packages/blocks/src/embed-block/image/image-block.ts

```diff
@@ -136,73 +135,26 @@ export class ImageBlockComponent extends ShadowlessElement {
   };
 
   @state()
-  private _imageState: 'waitUploaded' | 'loading' | 'ready' | 'failed' =
-    'loading';
-
-  private waitImageUploaded() {
-    return new Promise<void>(resolve => {
-      // If we could not get message from awareness in 1000ms,
-      // we assume this image is not found.
-      const timer = setTimeout(resolve, 2000);
-
-      const isBlobUploadingOnInit =
-        this.model.page.awarenessStore.isBlobUploading(this.model.sourceId);
-
-      const disposeSlot = this.model.page.awarenessStore.slots.update.on(() => {
-        const isBlobUploading = this.model.page.awarenessStore.isBlobUploading(
-          this.model.sourceId
-        );
-
-        /**
-         * case:
-         * clientA send image, but network latency is high,
-         * clientB got ydoc, but doesn't get awareness,
-         * clientC has a good network, and send awareness because of cursor changed,
-         * clientB receives awareness change from clientC,
-         * this listener will be called,
-         * but clientB doesn't get uploading state from clientA.
-         */
-        if (
-          isBlobUploadingOnInit === isBlobUploading &&
-          isBlobUploading === false
-        ) {
-          return;
-        }
-
-        if (!isBlobUploading) {
-          clearTimeout(timer);
-          resolve();
-        }
-      });
-
-      this._imageReady.dispose = () => {
-        disposeSlot.dispose();
-        clearTimeout(timer);
-        resolve();
-      };
-    });
-  }
+  private _imageState: 'loading' | 'ready' | 'failed' = 'loading';
 
   override async firstUpdated() {
     this.model.propsUpdated.on(() => this.requestUpdate());
     this.model.childrenUpdated.on(() => this.requestUpdate());
     // exclude padding and border width
     const { width, height } = this.model;
-    const storage = await this.model.page.blobs;
-    assertExists(storage);
+    const storage = this.model.page.blobs;
 
     this._imageState = 'loading';
-    let url = await storage.get(this.model.sourceId);
-    if (!url) {
-      this._imageState = 'waitUploaded';
-      await this.waitImageUploaded();
-      this._imageState = 'loading';
-      url = await storage.get(this.model.sourceId);
-    }
-    if (url) {
-      this._source = url;
-      this._imageState = 'ready';
-    } else {
+    try {
+      const blob = await storage.get(this.model.sourceId);
+      if (blob) {
+        this._source = URL.createObjectURL(blob);
```

Comment:
we may also need to revoke the blob url when the image is no longer being used.

Reply:
good catch